shard.lua

```diff
@@ -107,6 +107,22 @@ local function wait_connection()
     end
 end
 
+local function make_error(errno, fmt, ...)
+   if type(fmt) ~= 'string' then
+        fmt = tostring(fmt)
+    end
+
+    if select('#', ...) ~= 0 then
+        local stat
+        stat, fmt = pcall(string.format, fmt, ...)
+        if not stat then
+            error(fmt, 2)
+        end
+        fmt = tostring(fmt)
```

Comment:
doesn't `string.format` already return `string`? why do we need an extra conversion?

Reply:
not necessary, you're right.
but it's blank function if `type(fmt) == 'strint'`, otherwise (if something happened) it'll help.